india_compliance/gst_india/utils/gstin_info.py

```diff
@@ -92,6 +93,31 @@ def _extract_address_lines(address):
     return address_line1, address_line2
 
 
+@frappe.whitelist()
+def get_gst_category_from_gstin(gstin, country=None):
+    """
+    This function is used to guess GST Category from GSTIN.
+    Used where gstin info API is not available or used.
+    """
+    if not gstin:
+        if country in ("India", "", None):
+            return "Unregistered"
+
+        return "Overseas"
+
+    if TDS.match(gstin):
+        return "Tax Deductor"
+
+    if REGISTERED.match(gstin):
+        return "Registered Regular"
+
+    if UNBODY.match(gstin):
+        return "UIN Holders"
+
+    if OVERSEAS.match(gstin):
+        return "Overseas"
```

Comment:
this should happen in frontend itself - even if it means we need to duplicate some constants.